src/backend/columnar/columnar_tableam.c

```diff
@@ -983,8 +1070,149 @@ columnar_index_build_range_scan(Relation heapRelation,
 								void *callback_state,
 								TableScanDesc scan)
 {
-	ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-					errmsg("indexes not supported for columnar tables")));
+	/*
+	 * TODO: Should this function call pgstat_progress_update_param in
+	 * somewhere as heapam_index_build_range_scan ?
+	 */
+
+	if (start_blockno != 0 || numblocks != InvalidBlockNumber)
+	{
+		/* TODO: make sure that this is only valid for BRIN indexes */
+		ereport(ERROR, (errmsg("BRIN indexes on columnar tables are not supported")));
+	}
```

Comment:
have a test for that

Reply:
Not sure how I ended up with this assumption but `start_blockno` & `numblocks` seem to have nothing to do with `BRIN` indexes, but indeed with `CONCURRENT` index builds